src/cli/rustup_mode.rs

```diff
@@ -1006,21 +1006,19 @@ const DOCS_DATA: &[(&'static str, &'static str, &'static str,)] = &[
 fn doc(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
     let toolchain = explicit_or_dir_toolchain(cfg, m)?;
 
-    let doc_url = if let Some((_, _, path)) = DOCS_DATA
-        .into_iter()
-        .find(|(name, _, _)| m.is_present(name))
-    {
-        path
-    } else {
-        "index.html"
-    };
+    let doc_url =
+        if let Some((_, _, path)) = DOCS_DATA.iter().find(|(name, _, _)| m.is_present(name)) {
+            path
+        } else {
+            "index.html"
+        };
 
     if m.is_present("path") {
         let doc_path = toolchain.doc_path(doc_url)?;
         println!("{}", doc_path.display());
         Ok(())
     } else {
-        Ok(toolchain.open_docs(doc_url)?)
+        toolchain.open_docs(doc_url).map_err(|e| e.into())
```

Comment:
I don't think this change is work doing, and if we do I think you could use `map_err(Into::into)`